common/djangoapps/course_modes/tests/test_views.py

```diff
@@ -511,64 +516,118 @@ def test_course_closed(self):
+        self.assertContains(response, '<li class="collapsible-item">')
+        self.assertContains(response, 'access to all course activities')
+        self.assertContains(response, 'Full access')
+
+        # Check for happy path messaging - audit
+        self.assertContains(response, "discussion forums and non-graded assignments")
+        self.assertContains(response, "Get temporary access")
+        self.assertContains(response, "Access expires and all progress will be lost")
+
+    def _assert_unfbe_page(self, response, min_price=None, **_):
+        """
+        Assert track_selection.html and unfbe.html were transcluded.
+        """
+        # Check for string unique to track_selection.html.
+        self.assertContains(response, "| Upgrade Now")
+        # This string only occurs in lms/templates/course_modes/track_selection.html
+        # and related theme and translation files.
+
+        # Check for string unique to unfbe.html.
+        self.assertContains(response, "Some graded content may be lost")
+        # This string only occurs in lms/templates/course_modes/unfbe.html
+        # and related theme and translation files.
+
+        # Check min_price was correctly passed in.
+        self.assertContains(response, min_price)
+
+    def _assert_legacy_page(self, response, **_):
+        """
+        Assert choose.html was transcluded.
+        """
+        # Check for string unique to the legacy choose.html.
+        self.assertContains(response, "Choose Your Track")
+        # This string only occurs in lms/templates/course_modes/choose.html
+        # and related theme and translation files.
 
-        # Create the course modes and enroll the user
+    @ddt.data(
+        # gated_content_on, course_duration_limits_on, waffle_flag_on, expected_page_assertion_function
+        (True, True, True, _assert_fbe_page),
+        (True, False, True, _assert_unfbe_page),
+        (False, True, True, _assert_unfbe_page),
+        (False, False, True, _assert_unfbe_page),
+        (True, True, False, _assert_legacy_page),
+        (True, False, False, _assert_legacy_page),
+        (False, True, False, _assert_legacy_page),
+        (False, False, False, _assert_legacy_page),
+    )
+    @ddt.unpack
+    def test_track_selection_types(
+            self,
+            gated_content_on,
+            course_duration_limits_on,
+            waffle_flag_on,
+            expected_page_assertion_function
+    ):
+        """
+        Feature-based enrollment (FBE) is when gated content and course duration
+        limits are enabled when a user is auditing a course.
+
+        When prompted to perform track selection (choosing between the audit and
+        verified course modes), the learner may view 3 different pages:
+            1. fbe.html - full FBE
+            2. unfbe.html - partial or no FBE
+            3. choose.html - legacy track selection page
+
+        This test checks that the right page is shown.
```

Comment:
Might be worth specifying "...that the right template is rendered" since the URL will remain the same in all 3 cases.

Reply:
Done.